proposals/exp-show-target.md

```diff
@@ -0,0 +1,214 @@
+│ ff9e2fa        │ Nov 15, 2020 │       - │ 0.2           │ 20170428     │ 500                    │ 1                │ 20170428   │ 50                 │ -                  │ - │
+│ 11e1705        │ Nov 15, 2020 │       - │ 0.2           │ 20170428     │ 1500                   │ 2                │ 20170428   │ 10                 │ -                  │ - │
+│ a82585e        │ Nov 15, 2020 │       - │ 0.2           │ 20170428     │ 1500                   │ 2                │ 20170428   │ 10                 │ -                  │ - │
+│ c778a78        │ Nov 15, 2020 │       - │ 0.2           │ 20170428     │ 1500                   │ 2                │ 20170428   │ 10                 │ -                  │ - │
+│ 551082e        │ Nov 15, 2020 │       - │ 0.2           │ 20170428     │ 1500                   │ 2                │ 20170428   │ 10                 │ -                  │ - │
+│ 15bef96        │ Nov 15, 2020 │       - │ 0.2           │ 20170428     │ 1500                   │ 2                │ 20170428   │ 10                 │ -                  │ - │
+└────────────────┴──────────────┴─────────┴───────────────┴──────────────┴────────────────────────┴──────────────────┴────────────┴────────────────────┴────────────────────┴───┘
+```
+
+This table includes parameters and metrics from the `evaluate` stage and all
+stages upstream of it.  It's possible to also have a `--single-item` argument to
+exclude upstream changes, but this is out of scope for now.
+
+# How We Teach This
+
+See [Detailed design](#detailed-design) for a proposed help message.
+
+This is consistent with other commands, and it's only one additional argument to
+an existing command, so it likely doesn't require much teaching. It might be
+necessary to explicitly add that parameters and metrics from upstream stages
+will be shown.
+
+# Drawbacks
+
+This proposal may require changes to how DVC collects and tracks parameters and
+metrics. Implementation could break existing functionality, and it is
+inconsistent with the current behaviors of `dvc params`, `dvc metrics`, and `dvc
+exp diff` commands.
+
+Further, this proposal adds to an already crowded set of `dvc exp show` options,
+and there is no single way to show parameters and metrics that's going to work
+for every use case. For example, this proposal does nothing to filter out
+parameters for upstream stages, which may frequently be irrelevant if only the
+stages near the end of the pipeline change during experimentation.
+
+# Alternatives
+
+There are many alternative methods to refine the output of `dvc exp show`:
+
+* Current functionality: `exp show` already has options to include/exclude any
+  column, so the flexibility to get the same output as suggested here already
+  exists.
+* Expand the include/exclude options to accept wildcard/glob/regex arguments for
+  more flexibility to exclude an entire file or section of a file like
```

Comment:
Def. Another good proposal to include 👍

Reply:
> `exclude-params=params.json:local_config*`

This specific case may already be possible? (Without the wildcard) by specifying a group/parent param path.

Reply:
> Def. Another good proposal to include 👍 (or make an issue in the core repo directly?)

Added links to existing core dvc issues with some of these proposals.

> This specific case may already be possible? (Without the wildcard) by specifying a group/parent param path.

I don't think it works without a fully specified param name, but let me know if you find otherwise.

Reply:
Seems to work:

```console
$ dvc exp show --no-pager --include-params p
┏━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━┓
┃ Experiment    ┃ Created      ┃ p.one ┃ p.two ┃
┡━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━┩
│ workspace     │ -            │ 1     │ 2     │
│ master        │ Mar 01, 2021 │ 1     │ 2     │
│ ├── exp-4b280 │ Mar 01, 2021 │ 1     │ 2     │
│ └── exp-d662e │ Mar 01, 2021 │ 1     │ 2     │
└───────────────┴──────────────┴───────┴───────┘
$ dvc exp show --no-pager --include-params p.two
┏━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━┓
┃ Experiment    ┃ Created      ┃ p.two ┃
┡━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━┩
│ workspace     │ -            │ 2     │
│ master        │ Mar 01, 2021 │ 2     │
│ ├── exp-4b280 │ Mar 01, 2021 │ 2     │
│ └── exp-d662e │ Mar 01, 2021 │ 2     │
└───────────────┴──────────────┴───────┘
```

Reply:
Oh, nice! I had tried with just a filename, which doesn't work, but I guess a yaml section does work.

Reply:
The current include/exclude filtering implementation is naive - filtering is done at the UI level based on string comparisons for the column names.

Filtering by filename will work for cases where the table actually displays the filenames (meaning whenever a duplicate key exists in multiple files). But in normal cases where we do not need to display filenames, using `filename:` won't do anything.

So if your repo has `params1.json:foo` and `params2.json:foo`, the table will include the filenames as part of the column title, and `exp show` will also respect the `filename:` part of a column filter.

Reply:
See also https://github.com/iterative/dvc/pull/5720